ui/app/AppLayouts/Chat/panels/GroupChatPanel.qml

```diff
@@ -76,24 +82,30 @@ RowLayout {
          for (var i = 0; i < groupUsersModelListView.count; i ++) {
              var entry = groupUsersModelListView.itemAtIndex(i)
 
-             // Add all group users different than me
-             if(!entry.isAdmin) {
-                 d.groupUsersModel.insert(d.groupUsersModel.count,
+             d.groupUsersModel.insert(entry.isAdmin ? 0 : d.groupUsersModel.count,
                                       {"pubKey": entry.pubKey,
                                        "name": entry.name,
-                                       "icon": entry.icon})
-             }
+                                       "isAdmin": entry.isAdmin,
+                                       "tagIcon": entry.isAdmin ? "crown" : "",
+                                       "isReadonly": !root.rootStore.isCurrentUser(entry.pubKey)})
          }
 
          // Build contactsModel type from model type (to fit with expected StatusTagSelector format
          for (var j = 0; j < contactsModelListView.count; j ++) {
              var entry2 = contactsModelListView.itemAtIndex(j)
-             d.contactsModel.insert(d.contactsModel.count,
-                                {"pubKey": entry2.pubKey,
-                                 "displayName": entry2.name,
-                                 "icon": entry2.icon,
-                                 "isIdenticon": false,
-                                 "onlineStatus": false})
+             if (!entry2.isAdmin) {
+                 d.contactsModel.insert(d.contactsModel.count,
+                                        {"pubKey": entry2.pubKey,
+                                         "displayName": entry2.name,
+                                         "localNickname": !!entry2.localNickname ? entry2.localNickname : "",
+                                         "icon": entry2.icon,
+                                         "isAdmin": entry.isAdmin,
+                                         "isReadonly": !root.rootStore.isCurrentUser(entry.pubKey),
+                                         "isVerified": !!entry2.isVerified ? entry2.isVerified : false,
+                                         "isUntrustworthy": !!entry2.isUntrustworthy ? entry2.isUntrustworthy : false,
+                                         "isContact": !!entry2.isContact ? entry2.isContact : false,
```

Comment:
Why these !! checks are needed. May model not contain this data ?
If no maybe it's better to do checks at the beginning when filling item (line 67 ....) ?

Reply:
removed